test/e2e/gc_e2e_test.go

```diff
@@ -34,159 +33,268 @@ var _ = Describe("Garbage collector", func() {
 		operatorClient = ctx.Ctx().OperatorClient()
 	})
 
-	It("should delete a ClusterRole owned by a CustomResourceDefinition when the owner is deleted", func() {
-		group := fmt.Sprintf("%s.com", rand.String(16))
-		crd, err := kubeClient.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Create(context.TODO(), &apiextensionsv1.CustomResourceDefinition{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: fmt.Sprintf("plural.%s", group),
-			},
-			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
-				Group: group,
-				Scope: apiextensionsv1.ClusterScoped,
-				Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
-					{
-						Name:    "v1",
-						Served:  true,
-						Storage: true,
-						Schema: &apiextensionsv1.CustomResourceValidation{
-							OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{Type: "object"},
+	AfterEach(func() {
+		cleaner.NotifyTestComplete(GinkgoT(), true)
+	})
+
+	Context("Given a ClusterRole owned by a CustomResourceDefinition", func() {
+
+		When("CustomResourceDefinition is deleted", func() {
```

Comment:
I have the same feedback for this node. Specs under this context should be able to assume that a ClusterRole exists, and that the ClusterRole was at one point owned by a CRD, and that the CRD has since been deleted. Then the below spec can simply assert that the ClusterRole will eventually not exist.